Beam/resources/assets/js/remplib.js

```diff
@@ -572,6 +600,77 @@ remplib = typeof(remplib) === 'undefined' ? {} : remplib;
             if (0 === this.totalTimeSpent % logInterval) {
                 this.trackTimespent();
             }
+        },
+
+        actualProgress: function() {
+            var element = document.documentElement,
+                body = document.body;
+
+
+            return (body.scrollTop||body.scrollTop) / ((element.scrollHeight||body.scrollHeight - element.clientHeight));
+        },
+
+        scrollProgressEvent: function () {
+            var article = remplib.tracker.trackedArticle,
+                page = { page: remplib.tracker.actualProgress() },
+                currentPosition = (document.documentElement.scrollTop||document.body.scrollTop);
+
+            if (article) {
+                page.article = Math.min( 1, Math.max( 0, ( currentPosition - article.offsetTop) / article.offsetHeight ));
+            }
+
+            var event = document.createEvent('CustomEvent'); // old legacy way of defining custom events, because of IE support
```

Comment:
Toto nie je nutne. V zdielanej casti `remplib.js` je na toto nachystany uz polyfill. Mozes kludne zavolat 

```
 let event = new CustomEvent("foo", {
    bar: "baz",
});
window.dispatchEvent(event);
```

Podobne je to pouzite aj par riadkov vyssie.